app/assets/javascripts/privileges.js

```diff
@@ -27,7 +27,10 @@ $(() => {
     const $input = $td.find('.js-privilege-edit');
     const name = $input.data('name');
     const type = $input.data('type');
-    const value = parseFloat($input.val() || '') || null;
+
+    const rawValue = parseFloat($input.val() || '')
+
+    const value = rawValue || rawValue === 0 ? rawValue : null;
```

Comment:
Are the only possible falsy values for rawValue:
- 0
- NaN

If so, we could simplify this to:

```suggestion
    const value = Number.isNaN(rawValue) ? null : rawValue;
```

Reply:
Not a big fan of `Number.isNaN` (or the old `isNaN` function for that matter), but sure, I tried to come up with a scenario where it could fail, but every corner case seems to be correctly handled (and one has to deliberately attempt to work around the `Number.isNaN` check at that).

Reply:
If you wanted to avoid any type of `isNaN` the following is equivalent:

```js
    const value = rawValue === rawValue ? rawValue : null
```

I personally find this a little cryptic though - my preference is for being explicit about what's being checked for.